### DISCUSSION TITLE
Precision should support up to 18 decimal digits, and export value as string

### DISCUSSION THREAD

**Discussion by 8bu** [dm4t2/vue-currency-input#269] [Ideas]
For cryptocurrency input, sometimes it has more than 15 decimals place that need to be specific in the input. This precision's limitation make it hard to adapt these requirements. Can we support more than 15 decimal digits & export value as string (using BigNumber, or alternatives). This would be good when we can support these things. Thank you.

**Comment by dm4t2** [comment]
No, this is currently not possible because number values must be emitted as safe integers and a higher precision would result in inaccurate results (see #188). 

To support larger values or a higher precision, I need to evaluate if `BigInt` works as an internal value model. So emitted values might be expressed as `{value: 999999999999999999999999n, precision: 20}` for example.
